spec.md

```diff
@@ -600,6 +612,147 @@ message Volume {
+  // Indicates the volume is accessible only from the "region"
+  // "us-west1" and the "zone" "us-west1-b".
+  //
+  // Example 2:
+  // Given a volume that is accessible from two zones, and
+  // accessibility_requirements =
+  //   {"region": {"us-west1"}, "zone": {"us-west1-b", "us-west1-c"}}
+  // Indicates the volume is accessible from both "zone" "us-west1-b"
+  // and "zone" "us-west1-c" the "region" "us-west1".
+  map<string, ListOfString> accessibility_requirements = 4;
+}
+
+// TopologyRequirement captures the topological segment(s) the CO wants
+// a volume to be provisioned in.
+message TopologyRequirement {
+  // The topological segment(s) the provisioned volume MUST be
+  // accessible from.
+  // A topological segment is a sub-division of a cluster, like region,
+  // zone, rack, etc.
+  // Each key MUST consist of alphanumeric characters, '-', '_' or '.'.
+  // Each value MUST contain 1 or more valid strings.
+  // A value containing more than 1 string indicates a set of
+  // possibilities that the SP may choose from.
+  // An SP SHALL advertise the requirements for topology in
+  // documentation. COs SHALL permit passing through topological
+  // information.
+  // This field is OPTIONAL. If it is not specified, the SP MAY choose
+  // the topological segment(s) the provisioned volume is accessible
+  // from.
+  //
+  // Example 1:
+  // Given a volume that is accessible from a single zone, and
+  // accepted_topologies =
+  //   {"region": {"us-west1"}, "zone": {"us-west1-b"}}
+  // Indicates the volume MUST be accessible from the "region"
+  // "us-west1" and the "zone" "us-west1-b".
+  //
+  // Example 2:
+  // Given a volume that is accessible from a single zone, and
+  // accepted_topologies =
+  //   { "region": {"us-west1"}, "zone": {"us-west1-b", "us-west1-c"}}
+  // Indicates the volume may be provisioned in either the "zone"
+  // "us-west1-b" or the "zone" "us-west1-c" in the "region" "us-west1".
+  //
+  // Example 3:
+  // Given a volume that is accessible from two zones (which may be
+  // indicated via an opaque parameter if the SP supports both single
+  // and multi-zone volumes), and
+  // accepted_topologies =
+  //   {"region": {"us-west1"}, "zone": {"us-west1-b", "us-west1-c"}}
+  // Indicates the volume MUST be accessible from the "region"
+  // "us-west1" and both "zone" "us-west1-b" and "zone" "us-west1-c".
+  //
+  // Example 4:
+  // Given a volume that is accessible from two zones (which may be
+  // indicated via an opaque parameter if the SP supports both single
+  // and multi-zone volumes), and
+  // accepted_topologies =
+  //   {"region": {"us-west1"},
+  //    "zone": {"us-west1-b", "us-west1-c", "us-west1-d"}}
+  // Indicates the volume may be accessible from any combination of two
+  // "zones" in the "region" "us-west1": e.g. "us-west1-b" and
+  // "us-west1-c" or "us-west1-b" and "us-west1-d", etc.
+  map<string, ListOfString> accepted_topologies = 1;
```

Comment:
nit: avoid API stutter. The prefix of this type is already "Topology", we probably don't need it in the suffix as well. I would be perfectly happy with `accepted` or `requisite` or `mandatory` here.

Reply:
Good point, will remove suffix and leave prefixes only.